packages/core/src/servient.ts

```diff
@@ -70,7 +76,7 @@ export default class Servient {
     }
 
     public addThing(thing: ExposedThing): boolean {
-        if (thing.id === undefined) {
+        if (thing.id === undefined || thing.id === null || thing.id.length === 0) {
```

Comment:
As mentioned above, `== null` checks for both `undefined` and `null` at the same time.

```suggestion
        if (thing.id == null || thing.id.length === 0) {
```